Bootstrap release process

nikimanoledaki opened the pull request:
Closes #95 

This PR adds a release process that is the same as the one for the `ocm` CLI :)

Skarlso commented on .goreleaser.yaml:
```diff
@@ -0,0 +1,30 @@
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
```
We don't really support windows. For now I would take this out. The container fails starting and building on windows anyways. So that would probably fail the release.

Skarlso reviewed (approved):
One comment otherwise lgtm

phoban01 reviewed (commented):
Looks good. I'm wondering if we should consolidate the existing ghcr workflow into goreleaser? Let that handle all of the binary and images builds?

nikimanoledaki commented:
Thanks @phoban01, in the last commit I added the config needed for goreleaser to build and push the image. I used the login action that was in the existing ghcr workflow, which I deleted now. We should try a release to test it E2E and check if there's anything to debug.

nikimanoledaki commented:
@Skarlso, the last commit adds a big change so requesting a review again :)

phoban01 reviewed (approved):
LGTM. Thanks for adding docker @nikimanoledaki
